ratis-server/src/main/java/org/apache/ratis/server/impl/RoleInfo.java

```diff
@@ -47,7 +47,7 @@ class RoleInfo {
   public static final Logger LOG = LoggerFactory.getLogger(RoleInfo.class);
 
   private final RaftPeerId id;
-  private volatile RaftPeerRole role;
+  private final AtomicReference<RaftPeerRole> role = new AtomicReference<>();
```

Comment:
`volatile` and `AtomicReference` seems the same for this case?

Reply:
Thanks for the review, I will make changes

Reply:
Okay, I am fine to use `AtomicReference`.